### PR TITLE
Support rule appliedTo in Antrea NetworkPolicy

### PR DESCRIPTION
**Pull request by reachjainrahul**
With this PR, we can now support both RuleLevel AppliedTo and PolicyLevel AppliedTo configuration in Antrea NetworkPolicy.

### CODE REVIEW

**Comment by reachjainrahul** [comment]
/nephe-test-e2e-kind, /nephe-test-e2e-aws, /nephe-test-e2e-azure

File: pkg/controllers/networkpolicy/networkpolicy.go [hunk 1]
```diff
@@ -362,6 +377,12 @@ func (s *securityGroupImpl) deleteImpl(c cloudSecurityGroup, membershipOnly bool
 				s.deletePending = true
 				return nil
 			}
+		} else {
+			var err error
+			nps, err = r.networkPolicyIndexer.ByIndex(networkPolicyIndexerByAppliedToGrp, s.id.Name)
+			if err != nil {
+				return fmt.Errorf("get networkpolicy indexer with index=%v, name=%v: %w", networkPolicyIndexerByAppliedToGrp, s.id.Name, err)
```
Comment (Anandkumar26) [inline pkg/controllers/networkpolicy/networkpolicy.go:384]: nit: 
```suggestion
				return fmt.Errorf("get networkpolicy indexer with index=%v, name=%v: %v", networkPolicyIndexerByAppliedToGrp, s.id.Name, err)
```

**Comment by reachjainrahul** [comment]
/nephe-test-e2e-kind, /nephe-test-e2e-aks,  /nephe-test-e2e-eks

File: pkg/controllers/networkpolicy/networkpolicy.go (hunk 1 above)
Comment (reachjainrahul) [reply pkg/controllers/networkpolicy/networkpolicy.go]: Done

**Review by shenmo3 — DISMISSED** [review]
/LGTM

File: pkg/controllers/networkpolicy/networkpolicy.go (hunk 1 above)
Comment (shenmo3) [reply pkg/controllers/networkpolicy/networkpolicy.go]: Actually %w can and should be used for errors in format.
https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w

**Comment by reachjainrahul** [comment]
/nephe-test-e2e-kind, /nephe-test-e2e-aks, /nephe-test-e2e-eks

**Review by shenmo3 — APPROVED** [review]

**Pull request merged**